app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -87,6 +101,8 @@ const metaRPCClientFactory = (connectionStream) => {
           params,
           id,
         });
+        metaRPCClient.responseHandled = false;
+        metaRPCClient.timeoutAfterWait(cb);
```

Comment:
This method seems disconnected from the other things that MetaRPCClient is doing. The class already has `cb` because you're passing it when you call `object.requests.set(id, cb)`, so it seems like we ought to be able to use that. Perhaps we should extract the `requests.set` and `connectionStream.write` lines to a method and then place the `setTimeout` after that. This method could look something like:

``` javascript
class MetaRPCClient {
  // ...
  send(id, payload, cb) {
    this.requests.set(id, cb);
    this.connectionStream.write(payload);
    setTimeout(() => {
      if (!this.responseHandled) {
        cb(new Error('No response from RPC'), null);
      }
    }, 10000);
  }
  // ...
}
```

And now here in `metaRPCClientFactory` we could say:

``` javascript
object.send(id, {
  jsonrpc: '2.0',
  method: property,
  params,
  id,
}, cb);
```